sycl/source/detail/kernel_impl.cpp

```diff
@@ -57,12 +57,32 @@ kernel_impl::~kernel_impl() {
   }
 }
 
+template <info::kernel param>
+typename info::param_traits<info::kernel, param>::return_type
```

Comment:
I do not see this parameter in spec: https://www.khronos.org/registry/SYCL/specs/sycl-1.2.1.pdf

Reply:
This is just a fallback function to catch an unsupported parameter, now it also handles reference_count parameter.